Mage/src/main/java/mage/abilities/common/CastSpellPaidBySourceTriggeredAbility.java

```diff
@@ -71,7 +71,8 @@ public boolean checkTrigger(GameEvent event, Game game) {
         if (setTargetPointer) {
             this.getAllEffects().setTargetPointer(new FixedTarget(spell.getId(), game));
         }
+        this.getEffects().setValue("SpellCastId", spell.getId());
```

Comment:
It's better to use standard syntax like `spellCast` with full spell object (it's safe):

![shot_250620_122126](https://github.com/user-attachments/assets/de3d930a-c12d-4557-b65f-caf8bc108566)